### PR TITLE
PIM-5379: Remove completeness calculation from MongoDB saver

### PR DESCRIPTION
**Pull request by fitn**
Just some words to explain this PR.

3 issues are fixed here:

1. Useless completeness scheduling.
Completeness is overridden each time we save the product with this direct to db saver. No need to reschedule it, it will be done.

2. Wrong completeness generation.
The real saving part is done at the end of the `saveAll` method calling `insertDocuments` and `updateDocuments`. As long these methods are not called, the MongoDB collection current state is before the product update. In this case, the completeness is done on the "old" datas.

3. Memory leak on the product saver.
The rescheduling of the completeness is done with: `CompletenessGenerator::schedule(ProductInterface)`. It calls a clear to the collection. Accordingly, this will schedule the completeness to be removed in the UOW.
But now, we don't work anymore with the UOW here and there is no flush or clear of it. So products are detached from the UOW but completenesses are not and they are linked to products. Then products will be kept in memory.

This PR consists in removing the completeness scheduling and move the completeness calculation to call it after the product has been updated.

### CODE REVIEW

**Comment by nidup** [comment]
Behat CE : http://core-ci.akeneo.com/job/dci_behat/485/
Behat EE : http://core-ci.akeneo.com/job/dci_behat/486/

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Saver/ProductSaver.php [hunk 1]
```diff
@@ -105,14 +105,6 @@ public function saveAll(array $products, array $options = [])
             } else {
                 $productsToUpdate[] = $product;
             }
-
-            if (true === $options['schedule'] || true === $options['recalculate']) {
```
Comment (jmleroux) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Saver/ProductSaver.php]: You should not completly remove the use of `schedule`

**Comment by jmleroux** [comment]
GTM, good catch.
Thanks for this very clear description. For next time, you should also add the header array. See https://github.com/akeneo/pim-community-dev/blob/master/CONTRIBUTING.md for more info.

**Comment by nidup** [comment]
Rebase and GTM, thank you very much for the PR!

**Pull request merged**